sys/net/gnrc/network_layer/sixlowpan/iphc/gnrc_sixlowpan_iphc.c

```diff
@@ -1129,22 +1128,17 @@ static size_t _iphc_ipv6_encode(gnrc_pktsnip_t *pkt,
         iphc_hdr[IPHC2_IDX] |= SIXLOWPAN_IPHC2_M;
 
         /* if multicast address is of format ffXX::XXXX:XXXX:XXXX */
-        if ((ipv6_hdr->dst.u16[1].u16 == 0) &&
-            (ipv6_hdr->dst.u32[1].u32 == 0) &&
-            (ipv6_hdr->dst.u16[4].u16 == 0)) {
+        if (!memcpy(&ipv6_hdr->dst.u8[2], zeros8, 8)) {
```

Comment:
Just from reading the code its not clear where `zeros8` comes from, I'd rather compare byte-wise.

Reply:
So you're saying that this

```C
    if (!memcpy(ipv6_hdr->dest.u8[2], zeros8, 8)) {
        /* ... */
    }
```

is less readable than this

```C
    int is_all_zero = 1;
    for (size_t i = 2; i < 10; i++) {
        if (ipv6_hdr->dst.u8[2] != 0) {
            is_zero = 0;
            break;
        }
    }
    if (!is_all_zero) {
        /* ... */
    }
```

?

Reply:
no, but both your proposals are more obscure than

```C
if ((ipv6_hdr->dest.u8[2] == 0) && (ipv6_hdr->dest.u8[3] == 0) &&
    (ipv6_hdr->dest.u8[4] == 0) && (ipv6_hdr->dest.u8[5] == 0) &&
    (ipv6_hdr->dest.u8[6] == 0) && (ipv6_hdr->dest.u8[7] == 0) &&
    (ipv6_hdr->dest.u8[8] == 0) && (ipv6_hdr->dest.u8[9] == 0)) {
```

(I think even you reversed the logic in your second proposal ;-))